src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -109,42 +112,61 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
        shapes_it != tasks_to_dispatch_.end();) {
     auto &dispatch_queue = shapes_it->second;
     for (auto work_it = dispatch_queue.begin(); work_it != dispatch_queue.end();) {
-      auto work = *work_it;
-      auto task = std::get<0>(work);
-      auto spec = task.GetTaskSpecification();
+      auto &work = *work_it;
+      auto &task = std::get<0>(work);
+      auto &spec = task.GetTaskSpecification();
 
       std::shared_ptr<WorkerInterface> worker = worker_pool.PopWorker(spec);
       if (!worker) {
         // No worker available, we won't be able to schedule any kind of task.
         return;
       }
 
-      std::shared_ptr<TaskResourceInstances> allocated_instances(
-          new TaskResourceInstances());
-      bool schedulable = cluster_resource_scheduler_->AllocateLocalTaskResources(
-          spec.GetRequiredResources().GetResourceMap(), allocated_instances);
-      if (!schedulable) {
-        // Not enough resources to schedule this task.
+      const auto owner_worker_id = WorkerID::FromBinary(spec.CallerAddress().worker_id());
+      const auto owner_node_id = NodeID::FromBinary(spec.CallerAddress().raylet_id());
+      // If the owner has died since this task was queued, cancel the task by
+      // killing the worker (unless this task is for a detached actor).
+      if (!worker->IsDetachedActor() &&
+          !is_owner_alive_(owner_worker_id, owner_node_id)) {
+        // TODO (Alex): To @swang pls double check and make sure this is ok. <3 thx
+        RAY_LOG(ERROR) << "Owner of assigned task "
+                       << task.GetTaskSpecification().TaskId()
+                       << " died, killing leased worker " << worker->WorkerId();
```

Comment:
I don't think this error message is accurate. Could we use the term "caller" instead of "owner", since I think this is clearer to users? Also, maybe just say "canceling task" instead of "killing leased worker".

Reply:
Hmm yeah, I have a feeling this is one of those error messages that didn't evolve when the code around it did.

Updated.